### ISSUE TITLE
Nuclei Reporting GitHub Integration

### ISSUE THREAD

**Issue by hakrishi** [projectdiscovery/nuclei#3369]
Good afternoon, 

When using the -rc flag with the GitHub integration I would like an option to create 1 report/issue per scan instead of the default 1 report/issue per vulnerability. For example, lets say a scan has 30 vulnerabilities detected, at the moment it will create 1 issue per vulnerability whereas I'd like it to create 1 issue which contains all 30 vulnerabilities. 

In addition to this it would be nice to be able to update issues when new scans are run. Lets say I run a scan in Jan which finds 30 vulns and another scan in Feb which only finds 5 vulns. I'd like it to be able to update the existing issue that contains 30 vulns (perhaps with a new comment in the issue) with the newest detected vulnerabilities (5). 

Many thanks,

Rishi

**Comment by tarunKoyalwar** [comment]
@hakrishi , this is totally different implementation from current one . nuclei currently dedupes to avoid duplicate tickets on github . creating a report is complete different thing . a detailed nuclei report for 30 vulnerabilities with req and resp would be ~30 pages , which is not good & easy to read in github issue . but for some reason if you want this behaviour you can just generate markdown report and upload it to github

**Comment by hakrishi** [comment]
Also note I'd like the output format to be the same as a regular output file and for the vulnerabilities to be in 1 file/issue rather than multiple.

**Comment by tarunKoyalwar** [comment]
@hakrishi you can do this using `-me mdexport` . I think this answers your query
![Screenshot 2023-03-03 at 1 37 15 PM](https://user-images.githubusercontent.com/45962551/222666305-87419e97-c239-4cb4-9c00-07229615daf3.png)

**Comment by hakrishi** [comment]
thanks @tarunKoyalwar, I think I may be better off using the regular output (-o) file as this is the format I'm looking for. Do you know if it is possible to upload the output.txt file directly to a GitHub issue?

**Comment by tarunKoyalwar** [comment]
@ehsandeep , not sure about https://github.com/projectdiscovery/nuclei/issues/3369#issuecomment-1453167756 
do you think we should cover this in `nuclei` or `notify` ??

**Comment by hakrishi** [comment]
yeah sorry I think we went a bit off topic. I think it would be nice if all the vulnerabilities detected were created under 1 GitHub issue instead of multiple, not suggesting removing the current implementation but adding an option of sending all vulns to 1 issue instead of creating an issue for each vuln. Is this not possible?

**Comment by tarunKoyalwar** [comment]
@hakrishi , reporting client logic currently is executed per template and if we want to create 1 issues / (report/scan) that would require changes in codebase which may not be feasible . and also nuclei dedupes to avoid creating duplicate issues in github so not sure if it is possible to implement this . instead I think we can create a feature request for github support in `notify`
